### ISSUE TITLE
Deprecate MultiField()

### ISSUE THREAD

**Issue by mausch** [elastic/elasticsearch-net#1600]
I think `MultiField()` for mapping should be marked as `[Obsolete]`. It's been [deprecated since ES 1.0](https://www.elastic.co/guide/en/elasticsearch/reference/1.3/_multi_fields.html) so NEST should reflect that. 
Otherwise people will inadvertently use it (at least I did!) which causes problems down the line. In my particular case, this caused issues when comparing mappings.
Another solution would be generating the new-style multi-fields from the same `MultiField()` definition, but that would be a breaking change and could be confusing.

**Comment by Mpdreamz** [comment]
++ 

in NEST 2.0 they have been removed altogether.

**Comment by TheFireCookie** [comment]
What is this assigned for then? :)

+ What is the equivalent syntax on NEST for the "new" Fields of ES?

**Comment by gmarz** [comment]
@TheFireCookie use the [Fields()](https://github.com/elastic/elasticsearch-net/blob/develop/src/Nest/Domain/Mapping/Descriptors/StringMappingDescriptor.cs#L130) method under the property you're mapping:

```csharp
.String(s => s
    .Name("mainfield")
    .Fields(f => f
        .String(fs => fs.Name("mainfield.raw").Index(FieldIndexOption.NotAnalyzed))
    )
)
```

**Comment by TheFireCookie** [comment]
So I just that the .MultiField method to Obsolete and I can do a pull request  ?

**Comment by gmarz** [comment]
@TheFireCookie that's correct- a PR would be much appreciated!

**Comment by gmarz** [comment]
Closed via #1618
